src/ctapipe/irf/irfs.py

```diff
@@ -0,0 +1,385 @@
+    """Base class for irf parameterizations binned in true energy."""
+
+    true_energy_min = AstroQuantity(
+        help="Minimum value for True Energy bins",
+        default_value=u.Quantity(0.015, u.TeV),
+        physical_type=u.physical.energy,
+    ).tag(config=True)
+
+    true_energy_max = AstroQuantity(
+        help="Maximum value for True Energy bins",
+        default_value=u.Quantity(150, u.TeV),
+        physical_type=u.physical.energy,
+    ).tag(config=True)
+
+    true_energy_n_bins_per_decade = Integer(
+        help="Number of edges per decade for True Energy bins",
+        default_value=10,
+    ).tag(config=True)
+
+    def __init__(self, parent, **kwargs):
+        super().__init__(parent=parent, **kwargs)
+        self.true_energy_bins = create_bins_per_decade(
+            self.true_energy_min.to(u.TeV),
+            self.true_energy_max.to(u.TeV),
+            self.true_energy_n_bins_per_decade,
+        )
+
+
+class IrfRecoEnergyBase(Component):
+    """Base class for irf parameterizations binned in reconstructed energy."""
+
+    reco_energy_min = AstroQuantity(
+        help="Minimum value for Reco Energy bins",
+        default_value=u.Quantity(0.015, u.TeV),
+        physical_type=u.physical.energy,
+    ).tag(config=True)
+
+    reco_energy_max = AstroQuantity(
+        help="Maximum value for Reco Energy bins",
+        default_value=u.Quantity(150, u.TeV),
+        physical_type=u.physical.energy,
+    ).tag(config=True)
+
+    reco_energy_n_bins_per_decade = Integer(
+        help="Number of edges per decade for Reco Energy bins",
+        default_value=10,
+    ).tag(config=True)
+
+    def __init__(self, parent, **kwargs):
+        super().__init__(parent=parent, **kwargs)
+        self.reco_energy_bins = create_bins_per_decade(
+            self.reco_energy_min.to(u.TeV),
+            self.reco_energy_max.to(u.TeV),
+            self.reco_energy_n_bins_per_decade,
+        )
+
+
+class Irf2dBase(Component):
+    """Base class for radially symmetric irf parameterizations."""
+
+    fov_offset_min = AstroQuantity(
+        help="Minimum value for FoV Offset bins",
+        default_value=u.Quantity(0, u.deg),
+        physical_type=u.physical.angle,
+    ).tag(config=True)
+
+    fov_offset_max = AstroQuantity(
+        help="Maximum value for FoV offset bins",
+        default_value=u.Quantity(5, u.deg),
+        physical_type=u.physical.angle,
+    ).tag(config=True)
+
+    fov_offset_n_bins = Integer(
+        help="Number of FoV offset bins",
+        default_value=1,
+    ).tag(config=True)
+
+    def __init__(self, parent, **kwargs):
+        super().__init__(parent=parent, **kwargs)
+        self.fov_offset_bins = u.Quantity(
+            np.linspace(
+                self.fov_offset_min.to_value(u.deg),
+                self.fov_offset_max.to_value(u.deg),
+                self.fov_offset_n_bins + 1,
+            ),
+            u.deg,
+        )
+
+
+class PsfIrfBase(IrfTrueEnergyBase):
```

Comment:
I'd not call these classes "PsfIrf" or similar. They don't full fill the *is a* criterion.... They *are* not IRFs, they *compute* IRFs. 

So I'd call them something like `PSF3DMaker`